client/question/services/StdOutSeparatorService.js

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+ //
+ // Licensed under the Apache License, Version 2.0 (the "License");
+ // you may not use this file except in compliance with the License.
+ // You may obtain a copy of the License at
+ //
+ //      http://www.apache.org/licenses/LICENSE-2.0
+ //
+ // Unless required by applicable law or agreed to in writing, software
+ // distributed under the License is distributed on an "AS-IS" BASIS,
+ // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ // See the License for the specific language governing permissions and
+ // limitations under the License.
+
+ /**
+  * @fileoverview Service for displaying printed output.
+  */
+
+tie.factory('StdOutSeparatorService', [
+  function() {
+    /**
+     * The separator used in between test cases' stdOut.
+     *
+     * @type {string}
+     **/
+    var separator = "";
```

Comment:
Start with this being null, rather than an empty string.

Change setSeparator to init() (if it's only ever set once) or resetSeparator() (if it's set multiple times during a question session). That makes it clearer that it's supposed to be more of a "do once and then don't do it again" action.

Reply:
StdOutSeparatorService removed.